core/src/com/unciv/logic/battle/BattleDamage.kt

```diff
@@ -222,12 +226,14 @@ class BattleDamage{
     fun calculateDamageToAttacker(attacker: ICombatant, defender: ICombatant): Int {
         if(attacker.isRanged()) return 0
         if(defender.getUnitType().isCivilian()) return 0
-        val ratio = getDefendingStrength(attacker,defender) / getAttackingStrength(attacker,defender)
-        return (ratio * 30 * getHealthDependantDamageRatio(defender)).toInt()
+        val ratio = getAttackingStrength(attacker,defender) / getDefendingStrength(attacker,defender)
+        val i = if (ratio < 1) -1 else 1
+        return ((24 + 12 * Random().nextFloat()) * ((ratio.pow(i) + 3).pow(4) / 512 + 0.5).pow(i) * getHealthDependantDamageRatio(defender)).toInt()
```

Comment:
Where did this calculation come from? 0_0

Reply:
http://cache.baiducontent.com/c?m=9f65cb4a8c8507ed4fece76310508c31490797634b87834e29938448e435061e5a24feb82c291b01d8c57e610bb84828a9b32172414174a09bbece17cabae46a7ecb6423706bd3014f945bbb8e1b65972f8729eab86990adf14284d8d7c4ac2744bb22120ef0e78a2a1767c978806026e2d6c819025e60ad9c3b728e2d6028ef3436b4508ae3251e009680ac4b4db3619b7610e7b845b72963b304a56f0c5444b73cc71f212527944530fe342a7591fc29e23d783034c04fa4c9b7a19a3b9ba8ed26998adcdf248210edd29b8d663500&p=8a6c8e15d9c113ed01be9b7c4d&newp=84759a478adb17fc57efcb645853d8304a02c70e3ac3864e1290c408d23f061d4862e7b12525170ed0c37b6307a94c56e1f5327123454df6cc8a871d81edce3079&user=baidu&fm=sc&query=%A1%BE%D7%AA%CC%FB%A1%BFCC%C9%CF%D0%A1%B5%DCYY%B5%C4%B2%BD%B1%F8%C1%F7%D5%BD%CA%F5%B9%B9%CF%EB&qid=a4b3d95b002ead9a&p1=10&nsukey=Yw8GQO0TYyh%2BeLOAkzcpT8yGlI4CauuKAwO%2BFwJAgEJHRxcwNHoz9OAiRZ%2BDltGCSDxbsm9eV%2Fwxybn7qLx1%2Bh6ydh9sHnh70DYyA8csm8Evrq187S5Zim7Izu0Lw4mtNSKR6SC4z50A3QA2Zh%2F7K6v1aFXGhTFnlDnzmKuYRqtA0w8PdTBvtgqi6SoBMHqjp16%2BFqK4xE1FECo%2Fvhr59w%3D%3D

Reply:
The link doesn't work for me.
Have you played a game with this? Does the damage look OK?

Reply:
https://www.zhihu.com/question/321086400
it also mention that.

Reply:
Isn't that Civ VI, not V?

Also, this seems to have it at a different. and much simple, formula:

https://forums.civfanatics.com/threads/hans-lemurson-figures-out-the-combat-formula.606147/

Reply:
this method is that attack can do damage(if A>D) : (24+12P)*(（A/D+3^4/512+0.5)

Reply:
civ 6 and civ 5 is different.

Reply:
https://forums.civfanatics.com/threads/getting-the-combat-damage-math.646582/#post-15468029
it's an english article describe the Combat Formula about civ v, you can check what Legen said in #2. 
but he gives  30 as basic damage, in game it will ± 20%.